tests/agent_server/deploy/e2e/test_dryrun_new.py

```diff
@@ -0,0 +1,563 @@
+    assert undep == ["test::Resource[agent2,key=key5]", "test::Resource[agent2,key=key6]"]
+
+    result = await client.dryrun_trigger(uuid.uuid4(), version)
+    assert result.code == 404
+    result = await client.dryrun_trigger(environment, 123456789)
+    assert result.code == 404
+
+    result = await client.list_dryruns(uuid.uuid4(), version)
+    assert result.code == 404
+
+    result = await client.list_dryruns(environment, 123456789)
+    assert result.code == 404
+
+    result = await client.list_dryruns(environment, version)
+    assert result.code == 200
+    assert len(result.result["data"]) == 0
+
+    # request a dryrun with correct parameters
+    result = await client.dryrun_trigger(environment, version)
+    assert result.code == 200
+    dry_run_id = result.result["data"]
+
+    # get the dryrun results
+    result = await client.list_dryruns(environment, version)
+    assert result.code == 200
+    assert len(result.result["data"]) == 1
+
+    async def dryrun_finished():
+        result = await client.list_dryruns(environment, version)
+        logger.info("%s", result.result)
+        return result.result["data"][0]["todo"] == 0
+
+    await retry_limited(dryrun_finished, 10)
+
+    result = await client.get_dryrun_diff(uuid.uuid4(), version, dry_run_id)
+    assert result.code == 404
+    result = await client.get_dryrun_diff(environment, version, uuid.uuid4())
+    assert result.code == 404
+    result = await client.get_dryrun_diff(environment, version, dry_run_id)
+    assert result.code == 200
+    assert len(result.result["data"]["diff"]) == len(resources)
+    changes = result.result["data"]["diff"]
+
+    assert changes[0]["status"] == "added"
+    assert changes[0]["attributes"]["value"] == {
+        "from_value": None,
+        "from_value_compare": "",
+        "to_value": resources[0]["value"],
+        "to_value_compare": resources[0]["value"],
+    }
+
+    assert changes[1]["status"] == "modified"
+    assert changes[1]["attributes"]["value"] == {
+        "from_value": "incorrect_value",
+        "from_value_compare": "incorrect_value",
+        "to_value": resources[1]["value"],
+        "to_value_compare": resources[1]["value"],
+    }
+    assert changes[2]["status"] == "deleted"
+    assert changes[3]["status"] == "modified"
+    assert changes[3]["attributes"]["value"] == {
+        "from_value": {"first_level": {"nested": [5, 3, 2, 1, 1]}},
+        "from_value_compare": json.dumps({"first_level": {"nested": [5, 3, 2, 1, 1]}}, indent=4, sort_keys=True),
+        "to_value": resources[3]["value"],
+        "to_value_compare": json.dumps(resources[3]["value"], indent=4, sort_keys=True),
+    }
+    assert changes[4]["status"] == "unmodified"
+    assert changes[5]["status"] == "undefined"
+    assert changes[6]["status"] == "skipped_for_undefined"
+    assert changes[7]["status"] == "skipped_for_undefined"
+
+    pytest.skip("Agent down will only work when we have pause ability!!!")
```

Comment:
Do we have a ticket to remove this line when we have the pause ability? Or is it only a `FIXME`?

Reply:
We should indeed make sure we have no more skips at the end, I'll ad it the ticket.